lib/scenario_runner.py

```diff
@@ -319,7 +318,7 @@ def _checkout_repository(self):
             if problematic_symlink := utils.find_outside_symlinks(self._repo_folder):
                 raise RuntimeError(f"Repository contained outside symlink: {problematic_symlink}\nGMT cannot handle this in URL or Cluster mode due to security concerns. Please change or remove the symlink or run GMT locally.")
         else:
-            if self._branch:
+            if self._original_branch is not None:
```

Comment:
I see you introduced this new variable, but what is the reason here?

The code logic does not allow for the branch to be specified differently as what the user provided. The only way for the self._branch to be unset and then later be populated differently is through the route of the local directory. 
So my question: Did you run into any issues with the current implementation? If not I do not see the need for this extra var.

Reply:
Sorry, just read your comment about it in the conversation directly and now see the need. Is resolved.